ktor-server/ktor-server-plugins/ktor-server-di/common/src/io/ktor/server/plugins/di/DependencyInjection.kt

```diff
@@ -137,13 +159,15 @@ public val DI: ApplicationPlugin<DependencyInjectionConfig> =
                 }
             }
             monitor.subscribe(ApplicationStopping) {
-                for (key in map.keys.reversed()) {
+                for (key in dependencyMap.keys.reversed()) {
                     try {
                         val instance = registry.getDeferred<Any?>(key).tryGetCompleted() ?: continue
                         registry.shutdownHooks[key]?.invoke(instance)
                         onShutdown(key, instance)
                     } catch (e: Throwable) {
-                        environment.log.warn("Exception during cleanup for $key; continuing", e)
+                        if (e !is CancellationException) {
+                            environment.log.warn("Exception during cleanup for $key; continuing", e)
+                        }
```

Comment:
Should we call `ensureActive()` or we ignore `CancellationException` intentionally?
```suggestion
                        if (e !is CancellationException) {
                            environment.log.warn("Exception during cleanup for $key; continuing", e)
                        } else {
                            currentCoroutineScope().ensureActive()
                        }
```

Reply:
It's actually not inside a coroutine scope here, the shutdown hook is a blocking operation.

I'm actually noticing now that the shutdown hooks don't get ran when closing the process in the IDE.  Looks like Gradle just does a `kill -9` now.  Investigating...

Reply:
Turns out it's just stdout that is detached before the shutdown hooks are ran for any running gradle tasks.